src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -226,12 +226,15 @@ public final Throwable commit(Throwable accumulate)
     {
         try
         {
-            return txnProxy.commit(accumulate);
+            observers.forEach(SSTableFlushObserver::complete);
         }
-        finally
+        catch (Throwable t)
         {
-            observers.forEach(SSTableFlushObserver::complete);
+            // Return without advancing to COMMITTED, which will trigger abort() when the Transactional closes...
+            return Throwables.merge(accumulate, t);
         }
+
+        return txnProxy.commit(accumulate);
```

Comment:
The change here in some way mirrors what already happens for streaming and what now happens for import in this patch. We make sure index components are complete before proceeding w/ making the related SSTables visible for reads, etc.